### PR TITLE
Vulkan: Use compute shader for RealXFB decoding

### PR DESCRIPTION
**Pull request by stenzek**
Ends up being slightly faster (38% in a purely XFB upload bound scenario), but mainly trying to find a solution to the black screen/incorrect colours issue.

### CODE REVIEW

**Comment by JMC47** [comment]
@dolphin-emu-bot rebuild

**Comment by JMC47** [comment]
I didn't notice much of a performance difference, but if it's a performance up and realXFB works fine still, that's nice for me.

**Comment by degasus** [comment]
Is there a good way to share more code with the texture decoder of the vulkan backend?

**Comment by iwubcode** [comment]
@degasus - what if we shared this once AbstractTextures was merged?  We could have a function called DecodeFormatFromMemory() on the texture?  I think we'd have to fallback to CPU if there weren't any compute-shaders though.

This would fall in line with the conversation on IRC yesterday about hybrid-xfb and doing YCbCr to RGB to on the GPU if possible.

So we'd support converting both formats (YCbCr and YUYV) to RGB using either GPU or CPU?

**Comment by stenzek** [comment]
@degasus Yup, I don't see why it couldn't be added as another format for the texture decoding shaders.

**Comment by phire** [comment]
Hybrid XFB requires a XFB decoding shader anyway.

**Comment by iwubcode** [comment]
@phire - isn't that just another 'flavor' of converting some data to RGB format?

File: Source/Core/VideoBackends/Vulkan/TextureConverter.cpp [hunk 1]
```diff
@@ -811,36 +782,37 @@ bool TextureConverter::CompileYUYVConversionShaders()
...
     {
-      ivec2 uv = ivec2(gl_FragCoord.xy);
-      int buffer_pos = PC.buffer_offset + uv.y * PC.src_stride + (uv.x / 2);
-      vec4 c0 = texelFetch(source, buffer_pos);
+      uvec2 uv = gl_GlobalInvocationID.xy;
+      int buffer_pos = int(PC.buffer_offset + (uv.y * PC.src_stride) + (uv.x / 2));
+      vec4 yuyv = texelFetch(source, buffer_pos);
 
-      float y = mix(c0.r, c0.b, (uv.x & 1) == 1);
+      float y = mix(yuyv.r, yuyv.b, (uv.x & 1u) == 1u);
```
Comment (Tilka) [inline Source/Core/VideoBackends/Vulkan/TextureConverter.cpp]: mix() isn't cheap and I doubt compilers optimize this properly for booleans, just do a ternary if

**Comment by psennermann** [comment]
Is it possible to have a build link to test if this PR really fixes the black screen on Vulkan XFB?
Thanks

**Comment by JMC47** [comment]
Click show all tests and find the buildbot for whatever version you're using (Windows x64 for most people) and click details.  On that page you'll be able to find a download link to the test build.

**Comment by psennermann** [comment]
Oh thanks, unfortunately the black screen is still there...

**Comment by mimimi085181** [comment]
For me it did not fix the black screen issue. I have high hopes for https://github.com/dolphin-emu/dolphin/pull/5498 though.

**Comment by stenzek** [comment]
Irrelevant now that Hybrid XFB has been merged.

**Pull request closed without merging**